fili-generic-example/src/main/java/com/yahoo/wiki/webservice/data/config/dimension/GenericDimensions.java

```diff
@@ -0,0 +1,114 @@
+// Copyright 2016 Yahoo Inc.
+// Licensed under the terms of the Apache license. Please see LICENSE.md file distributed with this work for terms.
+package com.yahoo.wiki.webservice.data.config.dimension;
+
+import com.yahoo.bard.webservice.config.SystemConfig;
+import com.yahoo.bard.webservice.config.SystemConfigProvider;
+import com.yahoo.bard.webservice.data.config.dimension.DimensionConfig;
+import com.yahoo.bard.webservice.data.dimension.DimensionField;
+import com.yahoo.bard.webservice.data.dimension.KeyValueStore;
+import com.yahoo.bard.webservice.data.dimension.MapStoreManager;
+import com.yahoo.bard.webservice.data.dimension.SearchProvider;
+import com.yahoo.bard.webservice.data.dimension.impl.ScanSearchProviderManager;
+import com.yahoo.bard.webservice.util.EnumUtils;
+import com.yahoo.bard.webservice.util.StreamUtils;
+import com.yahoo.bard.webservice.util.Utils;
+import com.yahoo.wiki.webservice.data.config.auto.ConfigLoader;
+import com.yahoo.wiki.webservice.data.config.auto.DruidConfig;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.LinkedHashMap;
+import java.util.LinkedHashSet;
+import java.util.Set;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+/**
+ * Hold all the dimension configurations for the sample Bard instance.
+ */
+public class GenericDimensions {
+
+    private final SystemConfig systemConfig = SystemConfigProvider.getInstance();
+
+    private final String defaultDimensionBackendKey = systemConfig.getPackageVariableName("dimension_backend");
+
+    private final Set<DimensionConfig> dimensionConfigs;
+    private final LinkedHashMap<String, DimensionConfig> wikiApiDimensionNameToConfig;
+
+    /**
+     * Construct the dimension configurations.
+     */
+    public GenericDimensions(ConfigLoader configLoader) {
+        DruidConfig config = configLoader.getTableNames().get(0);
+        this.dimensionConfigs = Collections.unmodifiableSet(
```

Comment:
why are we taking something that supplies several DruidConfigs and then only taking the first one?
If we're only loading one, let's only pass in one.  If we're taking several, you should be looping over all the dimension configurations.